netmesh/model/netmesh/netmesh.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package netmesh;
+
+enum ApiVersion {
+    NETMESH_API_VERSION_1 = 0;
+};
+
+enum PayloadType {
+    NETMESH_PAYLOAD_IP = 0;
+    NETMESH_PAYLOAD_ETHERNET = 1;
+    NETMESH_PAYLOAD_MPLS = 2;
+    NETMESH_PAYLOAD_VXLAN = 3;
+    NETMESH_PAYLOAD_SRV6 = 4;
+};
+
+message NetworkServiceEndpoint {
+    string name = 1;
+    string uuid = 2;
+};
+
+message NetworkService {
+    string name = 1;
+    string uuid = 2;
+    string selector = 3;
+
+    message NetmeshChannel {
+        string name = 1;
+        PayloadType payload = 2;
+    };
+    repeated NetmeshChannel channels = 4;
+    ApiVersion version = 5;
```

Comment:
If we separate out the api version from this call, we can do some initial handshake to determine whether the client and service are capable of communicating with each other, rather than adding that to the data path. This would allow us to fail fast on initialization rather than on service request.

Reply:
See above.